Automattic/WP-Job-Manager
# Move from chosen library to select2 and remove deprecated asset enqueuing strategy

Fixes #1202 
Fixes #1358

See the issue for the reasons we're moving away from `chosen`.
#### Changes proposed in this Pull Request:

* Removes support for old asset enqueuing strategy deprecated in #1354.
* Deprecate the old `chosen` library. 
* Loads `select2` asset for category select and multiselect.

This will eventually also be used in the backend for background pulled select fields (#846).

#### Testing instructions:

* This will involve quite a bit of testing for compat with popular plugins and themes.
* Generally, make sure all frontend screens (all shortcodes) have the same formatting as `master` and the `<select />` fields get the fancy select2 treatment.


#### Proposed changelog entry for your changes:
* Developer: Moved from chosen JS library to select2.
